### PR TITLE
Update SBOM generation

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Update SBOM generation
  * Move cgmanifest into tools folder
  * Make cg detectiond a sbom use tools folder for component detection
  * Add package name and version
 
## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by kmccl922** [comment]
Harassment charges FBI people not playing stop the messages please
Sent from Yahoo Mail on Android 
 
  On Thu, Dec 16, 2021 at 7:13 PM, Travis ***@***.***> wrote:   
PR Summary

Update SBOM generation
   
   - Move cgmanifest into tools folder
   - Make cg detectiond a sbom use tools folder for component detection
   - Add package name and version

PR Context

PR Checklist
   
   -  PR has a meaningful title      
      - Use the present tense and imperative mood when describing your changes

   -  Summarized changes
   -  Make sure all .h, .cpp, .cs, .ps1 and .psm1 files have the correct copyright header
   -  This PR is ready to merge and is not Work in Progress.      
      - If the PR is work in progress, please add the prefix WIP: or [ WIP ] to the beginning of the title (the WIP bot will keep its status check at Pending while the prefix is present) and remove the prefix when the PR is ready.

   - Breaking changes      
      -  None
      - OR
      -  Experimental feature(s) needed         
         -  Experimental feature name(s): 


   - User-facing changes      
      -  Not Applicable
      - OR
      -  Documentation needed         
         -  Issue filed: 


   - Testing - New and feature      
      -  N/A or can only be tested interactively
      - OR
      -  Make sure you've added a new test if existing tests do not effectively test the code changed

   - Tooling      
      -  I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
      - OR
      -  I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:         
         -  Impact on PowerShell Editor Services which is used in the PowerShell extension for VSCode         
(which runs in a different PS Host).            
            -  Issue filed: 

         -  Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.            
            -  Issue filed: 

         -  Impact on PSScriptAnalyzer (which provides linting & formatting in the editor extensions).            
            -  Issue filed: 

         -  Impact on EditorSyntax (which provides syntax highlighting with in VSCode, GitHub, and many other editors).            
            -  Issue filed: 




You can view, comment on, or merge this pull request online at:

  https://github.com/PowerShell/PowerShell/pull/16641

Commit Summary
   
   - 65d362b switch to new sbom template
   - 5ec0f61 add names for linux
   - aebfbe6 move cgmanifest
   - 019e087 fix CG
   - e4e83c5 fix macos GC failures
   - c207fe8 use already set repoRoot for repoRoot
   - e0875c8 fix the correct template
   - 6f05fda fix autodetections
   - 9fa0df4 Revert "switch to new sbom template"

File Changes 
 
(22 files)
   
   -  M .github/prquantifier.yaml (2) 
   -  M .vsts-ci/linux.yml (2) 
   -  M .vsts-ci/mac.yml (2) 
   -  M .vsts-ci/misc-analysis/mdSpell.yml (2) 
   -  M .vsts-ci/windows.yml (2) 
   -  M .vsts-ci/windows/templates/windows-packaging.yml (1) 
   -  R tools/cgmanifest.json (0) 
   -  M tools/findMissingNotices.ps1 (4) 
   -  M tools/releaseBuild/azureDevOps/templates/SetVersionVariables.yml (22) 
   -  M tools/releaseBuild/azureDevOps/templates/compliance.yml (2) 
   -  M tools/releaseBuild/azureDevOps/templates/compliance/compliance.yml (2) 
   -  M tools/releaseBuild/azureDevOps/templates/compliance/generateNotice.yml (9) 
   -  M tools/releaseBuild/azureDevOps/templates/json.yml (2) 
   -  M tools/releaseBuild/azureDevOps/templates/linux.yml (20) 
   -  M tools/releaseBuild/azureDevOps/templates/mac-file-signing.yml (13) 
   -  M tools/releaseBuild/azureDevOps/templates/mac-package-build.yml (5) 
   -  M tools/releaseBuild/azureDevOps/templates/mac-package-signing.yml (13) 
   -  M tools/releaseBuild/azureDevOps/templates/mac.yml (2) 
   -  M tools/releaseBuild/azureDevOps/templates/nuget.yml (2) 
   -  M tools/releaseBuild/azureDevOps/templates/windows-hosted-build.yml (2) 
   -  M tools/releaseBuild/azureDevOps/templates/windows-package-signing.yml (12) 
   -  M tools/releaseBuild/azureDevOps/templates/windows-packaging.yml (5) 

Patch Links:
   
   - https://github.com/PowerShell/PowerShell/pull/16641.patch
   - https://github.com/PowerShell/PowerShell/pull/16641.diff

File: tools/releaseBuild/azureDevOps/templates/linux.yml [hunk 1]
```diff
@@ -148,6 +160,8 @@ jobs:
       value: false
     - name: NugetSecurityAnalysisWarningLevel
       value: none
+    - name: skipComponentGovernanceDetection
```
Comment (PaulHigin) [inline tools/releaseBuild/azureDevOps/templates/linux.yml:163]: I don't see where this variable is used...

File: tools/releaseBuild/azureDevOps/templates/mac-file-signing.yml [hunk 2]
```diff
@@ -10,19 +10,30 @@ jobs:
...
         value: false
       - name: NugetSecurityAnalysisWarningLevel
         value: none
+      - name: repoFolder
+        value: PowerShell
+      - name: repoRoot
+        value: $(Agent.BuildDirectory)\$(repoFolder)
+      - name: complianceRepoFolder
+        value: compliance
+      - name: complianceRepoPath
```
Comment (PaulHigin) [inline tools/releaseBuild/azureDevOps/templates/mac-file-signing.yml:26]: Variable does not seem to be used.

File: tools/releaseBuild/azureDevOps/templates/mac-package-signing.yml [hunk 3]
```diff
@@ -16,12 +16,23 @@ jobs:
       value: false
     - name: NugetSecurityAnalysisWarningLevel
       value: none
+    - name: repoFolder
+      value: PowerShell
+    - name: repoRoot
+      value: $(Agent.BuildDirectory)\$(repoFolder)
+    - name: complianceRepoFolder
+      value: compliance
+    - name: complianceRepoPath
```
Comment (PaulHigin) [inline tools/releaseBuild/azureDevOps/templates/mac-package-signing.yml:25]: Variable does not seem to be used.

File: tools/releaseBuild/azureDevOps/templates/windows-package-signing.yml [hunk 4]
```diff
@@ -15,13 +15,23 @@ jobs:
     - name: DOTNET_SKIP_FIRST_TIME_EXPERIENCE
       value: 1
     - group: ESRP
+    - name: repoFolder
+      value: PowerShell
+    - name: repoRoot
+      value: $(Agent.BuildDirectory)\$(repoFolder)
+    - name: complianceRepoFolder
+      value: compliance
+    - name: complianceRepoPath
```
Comment (PaulHigin) [inline tools/releaseBuild/azureDevOps/templates/windows-package-signing.yml:24]: Variable does not seem to be used.

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: tools/releaseBuild/azureDevOps/templates/linux.yml (hunk 1 above)
Comment (TravisEz13) [reply tools/releaseBuild/azureDevOps/templates/linux.yml:163]: it is used by Azure DevOps

File: tools/releaseBuild/azureDevOps/templates/mac-package-signing.yml [hunk 5]
```diff
@@ -16,12 +16,23 @@ jobs:
...
+      value: $(Agent.BuildDirectory)\$(complianceRepoFolder)
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/templates/mac-package-signing.yml:26]:
```suggestion
```

File: tools/releaseBuild/azureDevOps/templates/windows-package-signing.yml [hunk 6]
```diff
@@ -15,13 +15,23 @@ jobs:
...
+      value: $(Agent.BuildDirectory)\$(complianceRepoFolder)
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/templates/windows-package-signing.yml:25]:
```suggestion
```

File: tools/releaseBuild/azureDevOps/templates/mac-file-signing.yml [hunk 7]
```diff
@@ -10,19 +10,30 @@ jobs:
...
+        value: $(Agent.BuildDirectory)\$(complianceRepoFolder)
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/templates/mac-file-signing.yml:27]:
```suggestion
```

**Review by PaulHigin — APPROVED** [review]

**Pull request merged**